src/components/MerchantsPage/gam/CampaignProgressBar.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface Props {
+  numContributions: number;
+  targetAmount: number;
+  progressBarColor: string;
+  lastContributionTime: Date;
+  endDate: Date;
+}
+
+const CampaignProgressBar = ({
+  numContributions,
+  targetAmount,
+  progressBarColor,
+  lastContributionTime,
+  endDate,
+}: Props) => {
+  const timeSinceLastContribution = Date.now() - lastContributionTime.getTime();
+  const minutesSinceLastConstribution = timeSinceLastContribution / (1000 * 60);
+  const hoursSinceLastContribution = minutesSinceLastConstribution / 60;
+  const daysSinceLastContribution = hoursSinceLastContribution / 24;
+
+  const progressWidth = (raised: number, target: number) => {
+    if (raised < target) return (raised / target) * 100;
+    return 100;
+  };
+  const progressPercent = Math.floor((numContributions / targetAmount) * 100);
+
+  const timeUntilEnd = endDate.getTime() - Date.now();
+  const daysUntilEnd = timeUntilEnd / (1000 * 3600 * 24);
+
+  return (
+    <ProgressBarContainer>
+      {hoursSinceLastContribution < 1 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(minutesSinceLastConstribution)}m
```

Comment:
We should translate these static strings

Reply:
I feel like you can just have one line here instead of these three conditions
```suggestion
          <TimeStamp>
          Last contribution made {dateAsPresentable(this.lastContributionTime)} ago
        </TimeStamp>
```

where `dateAsPresentable` is some function that returns the right formatted string that's appropriate. That'll reduce the amount of duplicated code by a longshot here, and makes that date formatting reusable and easy to improve upon

Reply:
https://momentjs.com/ might actually be able to do just this for you

Reply:
Ok I wrote an anonymous function to handle the date presentation...thanks for the suggestion!